src/main/java/lotto/util/StringToIntConverter.java

```diff
@@ -0,0 +1,49 @@
+package lotto.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class StringToIntConverter {
+
+    private static final String DELIMITER = ",";
+
+    private StringToIntConverter() {
+    }
+
+    public static List<Integer> toInts(String input) {
```

Comment:
함수의 이름만 보고는 스트링을 복수개의 정수형으로 변환만 할거 같아요. 

이 과정에서 컴마(,)에 의해 구분되고 쪼개진다는 사실은 사용자가 파악할 수 없군요 😬

Reply:
메서드 네이밍을 개선하려고 하는데 한가지 고민이 있어서 질문드립니다.

```suggestion
    public static List<Integer> splitByCommaAndParseToInts(String input) {
```

함수 이름을 보고 의도를 정확하게 표현하려면 위와 같은 이름이 될 것 같은데 내부 구현을 과하게 드러낸다는 생각이 듭니다.

```suggestion
    public static List<Integer> toInts(List<String> input) {
```
이렇게 바꾸고 외부에서 스플릿 처리를 하고 넘겨주도록 바꾸려면 컨트롤러에서 변환처리를 하게 되는데 이건 또 컨트롤러의 역할은 아닌 것 같다는 생각이 드네요.

결국 **사용자 입력을 도메인 생성에 사용할 수 있도록 변환하는 책임을 누가 져야하는가?** 라는 고민으로 귀결되게 되는데요, 이걸 DTO를 적용해서 해결하는 것이 적절할지 궁금합니다! 대충 이렇게 구현될 것 같네요.

```java
public class WinningNumberDto {
    private final List<Integer> winningNumbers;
    
    public WinningNumberDto(String input) {
        winningNumbers = parseToInt(split(input));
    }
}

Reply:
제가 질문을 드린 의도는 함수의 이름만으로 함수가 하는 일을 충분하게 표현할 수 없기에 컴마로 구분되는지를 함수에 들어가봐야 알 수 있다는 점 때문이었어요. 어썸오도 이 의도를 이해하고 `splitByCommaAndParseToInts`라는 이름으로 수정해보셨네요! 

다만 우려하셨듯 구체적인 세부 구현이 함수의 이름을 통해 노출되어 있죠. 저라면 함수에 조금의 유연함을 주고, **사용자 입력을 구분하는 정책을 컨트롤러가 가지게** 하여 유틸리티 클래스가 정책을 담지 않도록 할거 같아요. 만일 다른 뷰에서 입력을 느낌표로 구분하는 경우가 있다면 유틸리티 메서드를 하나 더 만들기보다 정책에 따라 하나의 유틸리티 클래스를 재사용할 수 있도록 처리하는거죠. 

```java
List<Integer> input = StringToIntConverter.splitToInts(InputView.getWinningNumber(), LOTTO_INPUT_DELIMITER);
```

여기서 조금 더 확장하면 클래스명에서는 `Int`만 반환할거 같은데 메서드에서는 `Ints`를 반환하기에 클래스를 더 추상화하여 아래와 같이 변경할 수 있겠네요.

```java
List<Integer> input = StringConverter.splitToInts(InputView.getWinningNumber(), LOTTO_INPUT_DELIMITER);
```

---

컨트롤러에서 변환 처리를 해도 되는가!

어썸오는 생각을 더 나아가셨네요! 좋은 방향이라 생각해요. 뷰와 컨트롤러의 정의가 뭔지에 따라서 책임이 달라질거 같아요. 

어썸오가 생각하는 뷰와 컨트롤러의 역할은 무엇인가요?!

---

마지막으로 DTO! 

좋아요! DTO는 `레이어 간에` 데이터를 전달하기 위한 객체죠! DTO가 변환까지 해줘야하느냐, 요청하는 레이어(뷰)의 데이터를 그대로 전달해서 요청 받는 레이어(컨트롤러)에서는 그 값을 꺼내서 처리해줘야하느냐 의견이 많은걸로 알고 있어요. 그렇기에 꼭 이렇게 사용하자!는 없지만 저는 어썸오의 방식처럼 DTO가 변환해주는 방식을 사용한답니다. 그 이유는 DTO에 더러운 부분이 들어있기에 이를 사용하는 쪽에서는 코드가 깔끔해지기 때문이에요. 다만, DTO의 필드로 가지는 변수는 입력값을 그대로 받고 요청 받는 레이어가 DTO를 사용할 때에 변환해주는 방식을 사용해요. 이러면 입력 받은 기본 값을 상황에 맞게 변환해줄 수 있겠죠?

~~~java
public class WinningNumberDto {
    private final String winningNumbers;
    
    public WinningNumberDto(String input) {
        winningNumbers = input;
    }

    public List<Integer> toInts() {
        parseToInt(winningNumbers.split(","));
    }
}
~~~

DTO는 어디서든 사용될 수 있는 유틸리티 클래스와 달리 특정 레이어 간에서 사용되기 때문에 레이어간의 정책을(컴마로 구분한다) 들고 있을 수 있다고 생각해요.